x/concentrated-liquidity/swaps_test.go

```diff
@@ -3505,6 +3505,106 @@ func (s *KeeperTestSuite) TestComputeMaxInAmtGivenMaxTicksCrossed() {
 	}
 }
 
+// This test validates that swapping over the new extended price range in the lower
+// direction functions as expected. It validates that there are no unexpected failures
+// or panics when swapping to the new min tick and back.
+// Additionally, it validates that the swap amounts are roughly equal to the inverse amounts of a given swap.
+func (s *KeeperTestSuite) TestSwap_MinSpotPriceMigration() {
+
+	errTolerance := osmomath.ErrTolerance{
+		MultiplicativeTolerance: osmomath.MustNewDecFromStr("0.001"),
+	}
+
+	s.Run("out given in", func() {
+		s.SetupTest()
+
+		poolId := s.setupPositionsForMinSpotPriceMigration()
+
+		// Refetch pool
+		pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		// esimate amount in to swap left all the way until the new min initialized tick
+		amountZeroIn, _, _ := s.computeSwapAmounts(poolId, pool.GetCurrentSqrtPrice(), types.MinInitializedTickV2, true, false)
```

Comment:
Do you think it would be appropriate to also have a version that swaps to `MinCurrentTick`? Is this even possible?

Reply:
It should reach the `MinCurrentTick` if all liquidity is consumed in the full range and we cross `types.MinInitializedTick`.

This is expected to happen hear. Added assertions to make this clear:
[a2dbf41](https://github.com/osmosis-labs/osmosis/pull/6327/commits/a2dbf41fc94227def810879ba22f7a7eca40182d)